src/index.d.ts

```diff
@@ -0,0 +1,344 @@
+// Type definitions for found v0.3.5
+// Project: https://github.com/4Catalyzer/found
+// Definitions by: Kevin Ross <https://github.com/rosskevin/>
+declare module 'found' {
+  import * as React from 'react'
+
+  interface Map {
+    [key: string]: any
+  }
+
+  /**
+   * Location descriptor string:
+   *  store.dispatch(FarceActions.push('/foo?bar=baz#qux'));
+   *
+   * Equivalent location descriptor object:
+   *    store.dispatch(FarceActions.push({
+   *     pathname: '/foo',
+   *     search: '?bar=baz',
+   *     hash: '#qux',
+   *   }));
+   *
+   * https://github.com/4Catalyzer/farce#locations-and-location-descriptors
+   */
+  interface Location {
+    /**
+     * 'PUSH' or 'REPLACE' if the location was reached via FarceActions.push or FarceActions.replace respectively;
+     * 'POP' on the initial location, or if the location was reached via the browser back or forward buttons or
+     * via FarceActions.go
+     */
+    action: 'PUSH' | 'REPLACE' | 'POP'
+    /**
+     * the difference between the current index and the index of the previous location
+     */
+    delta: number
+    /**
+     * the location hash; as on window.location e.g. '#qux'
+     */
+    hash: string
+    /**
+     * the current index of the history entry, starting at 0 for the initial entry; this increments
+     * on FarceActions.push but not on FarceActions.replace
+     */
+    index: number
+    /**
+     * if present, a unique key identifying the current history entry
+     */
+    key?: string
+    /**
+     * the path name; as on window.location e.g. '/foo'
+     */
+    pathname: string
+    /**
+     * map version of search string
+     */
+    query: Map
+    /**
+     * the search string; as on window.location e.g. '?bar=baz'
+     */
+    search: string
+    /**
+     * additional location state that is not part of the URL
+     */
+    state: any
+  }
+
+  /**
+   * The shape might be different with a custom matcher or history enhancer, but the default matcher
+   * assumes and provides this shape. As such, this validator is purely for user convenience and
+   * should not be used internally.
+   */
+  interface Match {
+    location: Location
+    params: Map
+  }
+
+  /**
+   * An object implementing the matching algorithm.
+   *
+   * User code generally shouldn't need this, but it doesn't hurt to here, since we use it
+   * for routerShape below.
+   */
+  interface Matcher {
+    match: Function
+    getRoutes: Function
+    isActive: Function
+    /**
+     * Returns the path string for a pattern of the same format as a route path and a object of the
+     * corresponding path parameters
+     */
+    format: (pattern: any, params: Map) => any
+  }
+
+  /**
+   * Lenient arg version of location using in #push and #replace.
+   */
+  type LocationArg = Pick<Location, 'pathname'> &
```

Comment:
the union of this and "string" below is referred to as a "location descriptor" in the other docs: https://github.com/4Catalyzer/farce#locations-and-location-descriptors